src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+        }
+        deleteDriversByCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    private boolean deleteDriversByCar(Car car) {
+        String deleteAllDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement
+                            = connection.prepareStatement(deleteAllDriversQuery)) {
+            statement.setLong(1, car.getId());
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't deleted drivers by car " + car, e);
+        }
+    }
+
+    private Car addDriversToCar(Car car) {
+        String insertDriversQuery =
+                "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement statement
+                            = connection.prepareStatement(
+                                    insertDriversQuery)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers to car " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement deleteCarStatement
+                                = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectDrivers = "SELECT id"
+                + " FROM cars JOIN cars_drivers ON "
+                + "id = car_id"
+                + " WHERE driver_id = ? is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try {
+            Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement selectStatement
+                    = connection.prepareStatement(selectDrivers);
+            selectStatement.setLong(1, driverId);
+            ResultSet resultSet = selectStatement.executeQuery();
+            Car car = new Car();
+            selectStatement.close();
+            connection.close();
+            while (resultSet.next()) {
+                car = get(resultSet.getObject("id", Long.class)).get();
+                cars.add(car);
+            }
```

Comment:
https://stackoverflow.com/questions/25493837/java-cant-use-resultset-after-connection-close#:~:text=In%20your%20code%20you%20closed,calling%20method%20provides%20the%20Connection%20.